### PR TITLE
refactor(api): rework AssetScan General status

### PR DESCRIPTION
**Pull request by adamtagscherer**
## Description

This PR removes the `AssetScan.Status.General` property and moves it to `AssetScan.Status` one step higher.
Also it creates proper status states specific to the `AssetScan` API object and enforces it's state transitions when calling the API.

issue: https://github.com/openclarity/vmclarity/issues/664

## Type of Change

[ ] Bug Fix  
[ ] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: api/models/assetscanstatus.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		Message:            m,
+		LastTransitionTime: time.Now(),
+	}
+}
+
+func (as *AssetScanStatus) Equals(a *AssetScanStatus) bool {
+	if as.Message == nil && a.Message != nil {
+		return false
+	}
+	if a.Message == nil && as.Message != nil {
```
Comment (ramizpolic) [inline api/models/assetscanstatus.go:70]: nit: reusing the condition order makes it clearer
```suggestion
	if as.Message != nil && a.Message == nil {
```

**Review by ramizpolic — DISMISSED** [review]
Well done on the overall design and implementation, looks great! 🚀

File: api/openapi.yaml [hunk 2]
```diff
@@ -2418,6 +2418,79 @@ components:
...
+            | ----------- | -------------- | -------------------------------------------- |
+            | Pending     | Created        | AssetScan has been created                   |
+            | Scheduled   | Scheduled      | Moved to scheduled state                     |
+            | ReadyToScan | OutOfVMClarity | Started without VMClarity orchestration      |
+            | ReadyToScan | ReadyToScan    | Scans are ready to be started                |
+            | InProgress  | InProgress     | Scans are running on the target              |
+            | Aborted     | Aborted        | Scans have been aborted                      |
+            | Failed      | Failed         | At least one scan failed on target           |
+            | Failed      | AbortTimedOut  | AssetScan was in Aborted state for too long  |
+            | Done        | Done           | Scan finished successfully                   |
```
Comment (chrisgacsal) [inline api/openapi.yaml:2464]:
```suggestion
            | Pending     | Created        | Scan is pending as it has been newly created                  |
            | Scheduled   | Provisioning      | Scan is scheduled as Asset resources are being provisioned                  |
            | ReadyToScan | UnSupervised | Scan is created manually without the involvement of the Orchestrator (e.g. CI/CD scenario)     |
            | ReadyToScan | ReasourcesReady    | Scan is ready to scan as Asset resources have been provisioned |
            | InProgress  | ScannerIsRunning     | Scan is in progress as Scanner is still performing scanning on Asset resources             |
            | Aborted     | Cancellation       | Scan has been aborted due to cancellation request                     |
            | Aborted     | Timeout        | Scan has been aborted due to timeout                    |
            | Failed      | Error         | Scan has failed to to one or more errors reported by the Scanner           |
            | Failed      | AbortTimeout  | Scan has failed due to being in Aborted state too long  |
            | Done        | Success           | Scan has finished successfully                |
```

File: api/models/assetscanstatus.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+	if err := as.isValidStatusTransition(r); err != nil {
+		return err
+	}
+	if err := as.isValidReason(r); err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (chrisgacsal) [inline api/models/assetscanstatus.go:119]: Would be great to add a helper method which indicates if the state reached the end of the workflow.
Something like this:
```go
func (as *AssetScanStatus) Finished() bool {
	switch as {
        case AssetScanStatusReasonDone, AssetScanStatusReasonFailed:
            return true
        default:
            return false
}
```

**Review by chrisgacsal — CHANGES_REQUESTED** [review]

**Review by chrisgacsal — COMMENTED** [review]
LGTM! :shipit:

**Review by chrisgacsal — APPROVED** [review]

**Pull request merged**